content/inference/particlenet.md

```diff
@@ -1258,6 +1381,20 @@ The Tensorboard monitoring plots in the training/evaluation loss is shown as fol
 
 We see that the "heavy" model reaches even smaller training loss, meaning that the model does not meet the degradation issue yet. However, the evaluation loss is not catching up with the training loss, showing some degree of overtraining in this scheme. From the evaluation result, we see no improvement by moving to a heavy model.
 
+### 4. Apply preselection and class weights
+
+In HEP applications, it is sometimes required to train a multi-class classifier. While it is simple to specify the input classes in the `label` section of the `Weaver` data config, it is sometimes ignored to set up the preselection and assign the suitable class weights for training. Using an unoptimized configuration, the trained model will not reach the best performance although no error message will result. 
+
+Since our top tagging example is a binary classification problem, there is no specific need to configure the preselection and class weights. Below we summarize some experiences that may be applicable in reader's custom multi-class training task.
+
+The preselection should be chosen in a way that all remaining events passing the selection should fall into one and only one category. In other words, events with no labels attached should not be kept since it will confuse the training process.
+
+Class weights (the `class_weights` option under `weights` in the data config) control the relative importance of input sample categories for training. Implementation-wise, it changes the event possibility in a specific category chosen as training input events. The class weight comes into effect when one trains a multi-class classifier. Take 3-class case (denoted as [A, B, C]) as an example, the `class_weights: [1, 1, 1]` gives equal weights to all categories. Retraining the input with `class_weights: [10, 1, 1]` may result in a better discriminating power for class A vs. B or A vs. C; while the power of B separating with C will be weakened. As a trade-off between separating A vs. C and B vs. C, the class weights need to be intentionally tuned to achieve reasonable performance.
+
+Despite the different options for the class weights, one method to factor out the interplay across categories is to define "binarized" score between two classes. Suppose the raw score for the three classes are *P*(A), *P*(B), and *P*(C) (their sum should be 1), then one can define the discriminant *P*(BvsC) = *P*(B) / (*P*(B)+*P*(C)) to separate B vs. C. In this way, the saparating power of B vs. C will remain unchanged for `class_weights` configured as either `[1, 1, 1]` or `[10, 1, 1]`. This strategy has been widely used in CMS to define composite tagger discrimant which are applied analysis-wise.
```

Comment:
Did you mean to use the word "despite," which indicated that the next section can be done whether or not class weights are used? Or did you mean something like "Instead of using class weights, one method..."? That would indicate that this is a second option, instead of using class weights.

Reply:
Previously I thought it was "despite", but a better strategy is to apply both options. The sentence is changed